web-api/src/embedding.rs

```diff
@@ -12,68 +12,103 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+use std::str;
+
+use aws_config::retry::RetryConfig;
+use aws_sdk_sagemakerruntime::{config::Region, primitives::Blob, Client};
 use serde::{Deserialize, Serialize};
-use xayn_ai_bert::{AvgEmbedder, Config as EmbedderConfig, NormalizedEmbedding};
+use serde_json::json;
+use xayn_ai_bert::NormalizedEmbedding;
 
-use crate::{app::SetupError, error::common::InternalError, utils::RelativePathBuf};
+use crate::{app::SetupError, error::common::InternalError};
 
 #[derive(Debug, Deserialize, Serialize)]
 #[serde(default)]
 pub struct Config {
-    pub(crate) directory: RelativePathBuf,
-    pub(crate) token_size: usize,
+    pub(crate) sagemaker_endpoint_name: String,
+    pub(crate) aws_region: Option<String>,
+    pub(crate) aws_profile: Option<String>,
 }
 
 impl Default for Config {
     fn default() -> Self {
         Self {
-            directory: "assets".into(),
-            token_size: 250,
+            sagemaker_endpoint_name: String::new(),
+            aws_region: None,
+            aws_profile: None,
         }
     }
 }
 
+#[derive(Debug, Deserialize, Serialize)]
+struct Response {
+    embeddings: Vec<NormalizedEmbedding>,
+}
+
 pub(crate) struct Embedder {
-    embedder: AvgEmbedder,
+    embedding_dim: usize,
```

Comment:
:art: we usually call these kind of params with a `_size` suffix. `_dim` would be ok as well, we should just be consistent across the code base.

the main issue with `_dim` is that it is ambiguous, it could stand for dimensionality (which is what we actually refer to, like `384`), but it could also stand for dimension (which we don't refer to as it is the index of the model output array in which the embeddings lie, usually `2` for our models).